### PR TITLE
Cleanup docstrings for `gammapy.modeling`

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
This addresses part of issue https://github.com/gammapy/gammapy/issues/4811, specifically for `gammapy.modeling`

### CODE REVIEW

File: gammapy/modeling/models/core.py [hunk 1]
```diff
@@ -486,15 +484,15 @@ def write(
...
-        full_output : bool
+        full_output : bool, optional
             Store full parameter output. Default is False.
-        overwrite_templates : bool
+        overwrite_templates : bool, optional
             Overwrite templates FITS files. Default is False.
-        write_covariance : bool
-            Save covariance or not. Default is True.
+        write_covariance : bool, optional
+            Save the covariance. Default is True.
```
Comment (cgalelli) [inline gammapy/modeling/models/core.py:495]:
```suggestion
            Whether to save the covariance. Default is True.
```

File: gammapy/modeling/models/core.py [hunk 2]
```diff
@@ -702,26 +699,25 @@ def select(
...
+        name_substring : str, optional
+            Substring contained in the model name. Default is None.
+        datasets_names : str or list, optional
+            Name of the dataset. Default is None.
+        tag : str or list, optional
+            Model tag. Default is None.
+        model_type : {'None', 'spatial', 'spectral'}, optional
+           Type of model, used together with "tag", if the tag is not unique. Default is None.
+        frozen : bool, optional
+            Select models with all parameters frozen if True, exclude them if False.  Default is None.
```
Comment (cgalelli) [inline gammapy/modeling/models/core.py:715]:
```suggestion
            If True, select models with all parameters frozen; if False, exclude them. Default is None.
```

File: gammapy/modeling/parameter.py [hunk 3]
```diff
@@ -436,18 +436,20 @@ def _repr_html_(self):
...
-        """A deep copy"""
+        """Deep copy."""
         return copy.deepcopy(self)
 
     def update_from_dict(self, data):
-        """Update parameters from a dict.
-        Protection against changing parameter model, type, name."""
+        """Update parameters from a dictionary.
+
+        Protection against changing parameter model, type, name.
```
Comment (cgalelli) [inline gammapy/modeling/parameter.py:445]: This line is not very clear

**Review by cgalelli — COMMENTED** [review]
Thanks @Astro-Kirsty ! Usual inline comments...

**Review by cgalelli — DISMISSED** [review]

File: gammapy/modeling/models/core.py [hunk 4]
```diff
@@ -401,12 +399,12 @@ def covariance(self, covariance):
...
     @property
     def parameters(self):
-        """Parameters (`~gammapy.modeling.Parameters`)"""
+        """Parameters as a `~gammapy.modeling.Parameters` object."""
         return Parameters.from_stack([_.parameters for _ in self._models])
 
     @property
     def parameters_unique_names(self):
-        """List of unique parameter names as model_name.par_type.par_name"""
+        """List of unique parameter names as model_name.par_type.par_name."""
```
Comment (MRegeard) [inline gammapy/modeling/models/core.py:407]: This docstring was not really clear to me until I look at the methods. Maybe adding `output as model_name.par_type.par_name` would be more explanatory ? What do you think ?
```suggestion
        """List of unique parameter names output as a list of string formatted as model_name.par_type.par_name."""
```

File: gammapy/modeling/models/cube.py [hunk 5]
```diff
@@ -277,7 +276,7 @@ def contributes(self, mask, margin="0 deg"):
         Returns
         -------
         models : `DatasetModels`
-            Selected models contributing inside the region where mask==True
+            Selected models contributing inside the region where mask==True.
```
Comment (MRegeard) [inline gammapy/modeling/models/cube.py:279]:
```suggestion
            Selected models contributing inside the region where mask is set to True.
```

File: gammapy/modeling/models/cube.py [hunk 6]
```diff
@@ -715,16 +714,18 @@ def evaluate(self, energy, lon=None, lat=None):
             return value
 
     def copy(self, name=None, copy_data=False, **kwargs):
-        """Copy FoVBackgroundModel
+        """Copy FoVBackgroundModel.
```
Comment (MRegeard) [inline gammapy/modeling/models/cube.py:717]:
```suggestion
        """Copy the `FoVBackgroundModel` instance.
```

File: gammapy/modeling/models/spectral.py [hunk 7]
```diff
@@ -90,7 +90,7 @@ def scale_plot_flux(flux, energy_power=0):
 
 
 def integrate_spectrum(func, energy_min, energy_max, ndecade=100):
-    """Integrate 1d function using the log-log trapezoidal rule.
+    """Integrate 1 dimensional function using the log-log trapezoidal rule.
```
Comment (MRegeard) [inline gammapy/modeling/models/spectral.py:93]:
```suggestion
    """Integrate 1-dimensional function using the log-log trapezoidal rule.
```

File: gammapy/modeling/models/temporal.py [hunk 8]
```diff
@@ -314,11 +317,9 @@ class ExpDecayTemporalModel(TemporalModel):
     Parameters
     ----------
     t0 : `~astropy.units.Quantity`
-        Decay time scale
-        Default is 1 day
+        Decay timescale. Default is 1 day
```
Comment (MRegeard) [inline gammapy/modeling/models/temporal.py:320]:
```suggestion
        Decay timescale. Default is 1 day.
```

**Review by MRegeard — COMMENTED** [review]
Thank you very much @Astro-Kirsty, this one is massive !
See few inline comments.

**Review by MRegeard — DISMISSED** [review]

**Review by MRegeard — APPROVED** [review]

**Review by cgalelli — APPROVED** [review]

**Pull request merged**